### DISCUSSION TITLE
PDF A/3 Conformance

### DISCUSSION THREAD

**Discussion by faisalbwn** [QuestPDF/QuestPDF#785] [Q&A] [answered]
Hi,

Dear i need to create a PDF document with XML Attachment and with PDF A/3 Conformance Level. Can you please advice?

**Comment by faisalbwn** [comment]
@MarcinZiabek

**Comment by MarcinZiabek** [comment]
Unfortunately, file embedding is not available at this moment, and therefore generated documents do not target PDF A/3 version. 

I am revising the viability of introducing this feature in one of the following milestones by creating a wrapper around the `pdfium` library.

**Reply by faisalbwn** [reply]
Thanks. Looking forward to it.

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Since this thread was opened, both pieces you need have shipped. PDF/A-3 conformance can be enabled via [DocumentSettings](https://www.questpdf.com/concepts/document-settings.html) by setting `PDFA_Conformance = PDFA_Conformance.PDFA_3B`, and XML files can be embedded through the [Document Operations](https://www.questpdf.com/concepts/document-operations.html) API using `AddAttachment` with `Relationship = DocumentAttachmentRelationship.Alternative` for PDF/A-3b compliance. The documentation includes a worked example of exactly this XML-attachment-with-PDF/A-3b scenario.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
